common/src/main/java/com/microsoft/identity/common/internal/cache/SharedPreferencesFileManager.java

```diff
@@ -216,7 +244,7 @@ public final Map<String, String> getAll() {
                 final Map.Entry<String, String> entry = iterator.next();
                 //This is slightly wasteful, but we have no better key iterator and decryption
                 //is probably more painful than the additional file read when we miss in the cache.
-                String decryptedValue = getString(entry.getKey());
+                String decryptedValue = getStringInternal(entry.getKey(), entry.getValue());
```

Comment:
nit: im wondering if it would be cleaner to read if 

1. We created a "getValueFromSharedPrefEntry()" method, which performs both decryption and writeback.
2. Invoke the method here and in getString().

Reply:
I tried to figure out a way to make this work, but I don't think I can structure it this way without adding the overall flow making extraneous calls to SharedPreferences.getString(), which has a larger perf cost that just using the values we already got from SharedPreferences.getAll().

We always want to prefer existing decrypted-cache values over either using the getAll() value or a new SharedPreferences.getString() and decrypting + write-backing it, and to avoid getString() calls if that value exists.

Reply:
I could make the second value a Callable<String> and pass `() -> mSharedPreferences.getString(key, null)` for the getString() implementation and `() -> entry.getValue()` for the getAll() implementation.  Do you think that would be an improvement?